readthedocs/proxito/middleware.py

```diff
@@ -258,6 +168,56 @@ def add_cache_headers(self, request, response):
             # Set the key to private only if it hasn't already been set by the view.
             response.headers.setdefault('CDN-Cache-Control', 'private')
 
+    def _set_request_attributes(self, request, unresolved_domain):
+        """
+        Set attributes in the request from the unresolved domain.
+
+        - If the project was extracted from the ``X-RTD-Slug`` header,
+          we set ``request.rtdheader`` to `True`.
+        - If the project was extracted from the public domain,
+          we set ``request.subdomain`` to `True`.
+        - If the project was extracted from a custom domain,
+          we set ``request.cname`` to `True`.
+        - If the domain needs to redirect, set the canonicalize attribute accordingly.
+        """
+        # TODO: Set the unresolved domain in the request instead of each of these attributes.
+        origin = unresolved_domain.origin
+        project = unresolved_domain.project
+        if origin == OriginType.http_header:
+            request.rtdheader = True
```

Comment:
You have it on your list to get rid of these right, and just use the `unresolved_domain.origin` object in the next refactor, right? :)

Reply:
Yes, please 🙏🏼

Reply:
Already done in https://github.com/readthedocs/readthedocs.org/pull/9982/ :D